### ISSUE TITLE
Customer Managed Key user_assigned_identity is not Optional

### ISSUE THREAD

**Issue by thamjayshen** [Azure/terraform-azurerm-avm-res-storage-storageaccount#194]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Greenfield/Brownfield provisioning

greenfield

### Terraform Version

>=1.9.2

### Module Version

0.2.7

### AzureRM Provider Version

3.117.0

### Affected Resource(s)/Data Source(s)

azurerm_storage_account_customer_managed_key

### Terraform Configuration Files

```hcl
sa1= {
    name                      = "sa1"
    resource_group_name                    = "rg1"
    enable_telemetry          = true
    account_replication_type  = "ZRS"
    shared_access_key_enabled = true
    private_endpoints = {
      pep_flowlog = {
        name                            = "pep-blob-pmxnprdvnetfl01"
        subnet_resource_id              = "xxx"
        private_dns_zone_resource_ids   = ["/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.Network/privateDnsZones/privatelink.blob.core.windows.net"]
        subresource_name                = "blob"
        private_service_connection_name = "psc-blob-sa1"
      }
    }

    managed_identities = {
      system_assigned = "true"
    }

    customer_managed_key = {
      key_vault_resource_id = "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.KeyVault/vaults/kv-with-key"
      key_name              = "sa1-key"
    }

  }
```

### tfvars variables values

```hcl
Provided in the configuration file above.
```

### Debug Output/Panic Output

```shell
Error: Attempt to get attribute from null value on .terraform/modules/storage_accounts/main.tf line 327, in resource "azurerm_storage_account_customer_managed_key" "this":
327:   user_assigned_identity_id = var.customer_managed_key.user_assigned_identity.resource_id
var.customer_managed_key.user_assigned_identity is null
This value is null, so it does not have any attributes.
```

### Expected Behaviour

It should pass because user_assigned_identity_resource_id is optional parameter.

### Actual Behaviour

Error

### References

The block of code wrapped with ** ** causes issues in https://github.com/Azure/terraform-azurerm-avm-res-storage-storageaccount/blob/v0.2.9/main.tf:

```
resource "azurerm_storage_account_customer_managed_key" "this" {
  count = var.customer_managed_key != null ? 1 : 0

  key_name                  = var.customer_managed_key.key_name
  storage_account_id        = azurerm_storage_account.this.id
  key_vault_id              = var.customer_managed_key.key_vault_resource_id
  key_version               = var.customer_managed_key.key_version
  user_assigned_identity_id = **var.customer_managed_key.user_assigned_identity.resource_id**

  lifecycle {
    precondition {
      condition     = (var.account_kind == "StorageV2" || var.account_tier == "Premium")
      error_message = "`var.customer_managed_key` can only be set when the `account_kind` is set to `StorageV2` or `account_tier` set to `Premium`, and the identity type is `UserAssigned`."
    }
  }
}
```

Solution:

In the main.tf below is changed to:
```
user_assigned_identity_id = var.customer_managed_key.user_assigned_identity != null 
  ? var.customer_managed_key.user_assigned_identity.resource_id 
  : null

```

This checks for whether user_assigned_identity is defined first before looking into the resource_id

**Comment by chinthakaru** [comment]
@thamjayshen Thanks for reaching out. I am planning to make few changes soon and will release this within that change.

**Comment by chinthakaru** [comment]
@thamjayshen This has been fixed on v 0.3.0 release.
